clang/include/clang/Basic/Attr.td

```diff
@@ -3240,6 +3244,7 @@ def PreferredName : InheritableAttr {
   let InheritEvenIfAlreadyPresent = 1;
   let MeaningfulToClassTemplateDefinition = 1;
   let TemplateDependent = 1;
+  let DeferDeserialization = 1;
```

Comment:
Maybe it is better to add comment here to explain why we want to defer PreferredName.